domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/displaycontextparameter/DisplayContextParameterType.java

```diff
@@ -0,0 +1,48 @@
+package uk.gov.hmcts.ccd.definition.store.domain.displaycontextparameter;
+
+import com.google.common.base.Strings;
+
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public enum DisplayContextParameterType {
+    DATETIMEENTRY,
+    DATETIMEDISPLAY,
+    TABLE,
+    LIST;
+
+    private static final Pattern PATTERN = Pattern.compile("#(.+)\\((.+)\\)");
```

Comment:
can't we improve the Regex to list the supported types of DCP? i.e.
"#(TABLE|LIST|DATEENTRY|...)\\((.+)\\)"
just a suggestion not sure whether this would help simplify the code somewhere else

Reply:
Considered, overall doesn't make much difference to the code I don't think, and any new enums would need to be added to the regex too which adds a bit more maintenance. This way can also more easily handle when an invalid type (e.g. `#INVALIDTYPE(value)`) is passed.